internal/server/v1beta1/asset.go

```diff
@@ -211,6 +211,58 @@ func (server *APIServer) GetAssetByVersion(ctx context.Context, req *compassv1be
 	}, nil
 }
 
+func (server *APIServer) UpsertAsset(ctx context.Context, req *compassv1beta1.UpsertAssetRequest) (*compassv1beta1.UpsertAssetResponse, error) {
+	userID, err := server.validateUserInCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	baseAsset := req.GetAsset()
+	if baseAsset == nil {
+		return nil, status.Error(codes.InvalidArgument, "asset cannot be empty")
+	}
+
+	ast := server.buildAsset(baseAsset)
+	ast.UpdatedBy.ID = userID
```

Comment:
```
type upsertRequest interface {
	GetAsset() *compassv1beta1.UpsertPatchAssetRequest_BaseAsset
}
```

We can't use this because `GetAsset()` response would be different for `UpsertAssetRequest`. Also both APIs are using different statsd metric name.

Reply:
@sudo-suhas updated here
https://github.com/odpf/compass/pull/171/commits/70347453366d12b35aded46d0e5e5ef1654957fd